nbclient/client.py

```diff
@@ -442,13 +468,14 @@ async def async_setup_kernel(self, **kwargs):
         if self.km is None:
             self.start_kernel_manager()
 
-        if not self.km.has_kernel:
-            await self.async_start_new_kernel_client(**kwargs)
-        try:
-            yield
-        finally:
-            if cleanup_kc:
-                await self._async_cleanup_kernel()
+        with self.graceful_shutdown():
+            if not self.km.has_kernel:
+                await self.async_start_new_kernel_client(**kwargs)
+            try:
+                yield
+            finally:
+                if cleanup_kc:
```

Comment:
Probably can just move this into the graceful shutdown context wrapper now?

Reply:
good point. tbh I don't see a reason for graceful shutdown to be a separate function.

Reply:
I'm also intentionally not respecting cleanup_kc in case of unexpected shutdown.